examples/distributed/partition_graph.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+import os.path as osp
+
+import torch
+from ogb.nodeproppred import PygNodePropPredDataset
+
+from torch_geometric.distributed.partition.partitioner import (
+    Partitioner,
+    prepare_directory,
+)
+
+
+def partition_dataset(ogbn_dataset: str, root_dir: str, num_partitions: int):
+    save_dir = root_dir + "/partition"
+    dataset = PygNodePropPredDataset(ogbn_dataset, root_dir)
+    data = dataset[0]
+
+    partitioner = Partitioner(output_dir=save_dir, num_parts=num_partitions,
+                              data=data)
+    partitioner.generate_partition()
+    split_idx = dataset.get_idx_split()
+    print('-- Saving label ...')
+    label_dir = prepare_directory(root_dir, f'{ogbn_dataset}-label')
+    torch.save(data.y.squeeze(), osp.join(label_dir, 'label.pt'))
+
+    print('-- Partitioning training idx ...')
+    train_idx = split_idx['train']
+    train_idx = train_idx.split(train_idx.size(0) // num_partitions)
+    train_idx_partitions_dir = prepare_directory(
+        root_dir, f'{ogbn_dataset}-train-partitions')
+    for pidx in range(num_partitions):
+        torch.save(train_idx[pidx],
+                   osp.join(train_idx_partitions_dir, f'partition{pidx}.pt'))
+
+    print('-- Partitioning test idx ...')
+    test_idx = split_idx['test']
+    test_idx = test_idx.split(test_idx.size(0) // num_partitions)
+    test_idx_partitions_dir = prepare_directory(
+        root_dir, f'{ogbn_dataset}-test-partitions')
+    for pidx in range(num_partitions):
+        torch.save(test_idx[pidx],
+                   osp.join(test_idx_partitions_dir, f'partition{pidx}.pt'))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Arguments for ClusterData Partitioning.")
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default='ogbn-products',
+        help="The name of dataset.",
```

Comment:
The purpose of this example code is that we want to add an example for distributed training, and this partition example is part of this work. Since the total procedure is not ready, we can delete this example for now and add back it when all the components are ready.